### ISSUE TITLE
Why remove usage of tf.layers.dense and why change to use softmax_cross_entropy_with_logits_v2 ?

### ISSUE THREAD

**Issue by chenghuige** [HKUST-KnowComp/R-Net#49]
I see func.py code change from   
using tf.layers.dense to new written dense function in func.py, why to make such decision ?  
Seems they do the same thing.  
Also change from using softmax_cross_entropy_with_logits to softmax_cross_entropy_with_logits_v2, did it improve performance ? Seems if stop_gradient for label they are the same.

**Comment by wzhouad** [comment]
Tensor reuse in `tf.layers` is complex and error-prone, I reimplement it to avoid the bugs.
`softmax_cross_entropy_with_logits` is deprecated in tf1.5, `v2` is actually the same.

**Comment by chenghuige** [comment]
@wzhouad  Thanks!  closing
